### DISCUSSION TITLE
feature: Add extras.lang.NuShell

### DISCUSSION THREAD

**Discussion by gplusplus314** [LazyVim/LazyVim#3480] [Extra Requests]
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

Not really a problem, no. But LSP and CMP configuration is just a bit complicated.

### Describe the solution you'd like

I really love the concept of the `extras.lang.*` packages and it's my favorite aspect of LazyVim. It would be really great if we could add NuShell (the "nu" language) to LazyVim so that it "just works", like the other supported languages.

### Describe alternatives you've considered

Manually configuring the LSP and autocomplete. To be honest, I haven't been successful at this unless I use vanilla NeoVim from scratch - I always end up breaking something in LazyVim.

### Additional context

There's an official [nvim-lspconfig here](https://github.com/neovim/nvim-lspconfig/pull/2884).

**Comment by OpalBolt** [comment]
I think this got introduced in: https://www.lazyvim.org/news#12x :)

**Comment by AdrienLemaire** [comment]
It seems like lazyvim uses nvim-lspconfig fine ([nvim-lspconfig#nushell](https://github.com/neovim/nvim-lspconfig/blob/master/doc/configs.md#nushell), [lang.nushell](http://www.lazyvim.org/extras/lang/nushell))
the `gd` support is there.
jumps aren't nice though, `]f` doesn't work and navigating through `{` with `]m` is not pleasant.
Is it due to a lack of support by nvim-lspconfig, or am I missing something ?
